features/netsocket/NetworkStack.h

```diff
@@ -183,6 +183,8 @@ class NetworkStack: public DNS {
 
 protected:
     friend class InternetSocket;
+    friend class InternetDatagram;
+    friend class ICMPSocket;
```

Comment:
Not sure that `UDPSocket` and `ICMPSocket` actually need to be friends any more - `InternetDatagramSocket` does the work. Doesn't really matter though.